README.md

```diff
@@ -1,9 +1,201 @@
 # knuu
 
-This repository contains the source code for the knuu Integration Test Framework.
+[![CodeQL](https://github.com/celestiaorg/knuu/workflows/CodeQL/badge.svg)](https://github.com/celestiaorg/knuu/actions) [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0) [![OpenSSF Best Practices](https://bestpractices.coreinfrastructure.org/projects/7475/badge)](https://bestpractices.coreinfrastructure.org/projects/7475)
 
-If you have feedback on the framework, want to report a bug or suggest an improvement, please create an issue.
+## Description
 
-## Example
+Knuu is an open-source project, and the goal of knuu is to provide a framework for writing integration tests.
+The framework is written in Go and is designed to be used in Go projects.
+The idea is to provide a framework that uses the power of containers and Kubernetes without the test writer having to know the details of how to use them.
 
-Check out the [knuu-example](https://github.com/celestiaorg/knuu-example) repository for an example of how to use the knuu Integration Test Framework.
+We invite you to explore our codebase, contribute, and join us in developing a framework to help projects write integration tests.
+
+## Features
+
+Knuu is designed around `Instances`, which you can create, start, control, communicate with other Instances, stop, and destroy.
+
+Some of the features of knuu are:
+
+- Initialize an Instance from a Container/Docker image
+- Configure startup commands
+- Configure Networking
+    - What ports to expose
+    - Disable networking to simulate network outages
+- Configure Storage
+- Execute Commands
+- Configure resources
+- Create a pool of Instances and control them as a group
+- And more to come...
+
+> If you have feedback on the framework, want to report a bug, or suggest an improvement, please create an issue.
+
+## Getting Started
+
+This section will guide you on how to set up and run **knuu**.
+
+### Prerequisites
+
+1. **Docker**: Knuu requires Docker to run. You can install Docker by following the instructions [here](https://docs.docker.com/get-docker/).
+
+2. **Kubernetes cluster**: Set up access to a Kubernetes cluster using a kubeconfig.
+
+3. **'test' Namespace**: Create a namespace called 'test' in your Kubernetes cluster.
+   > **Note:** The used namespace can be changed by setting the `KNUU_NAMESPACE` environment variable.
```

Comment:
[non-blocking][proposal]: it won't hurt to provide more information on how to setup a cluster